packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -100,7 +100,7 @@ export interface UsePickerValueBaseProps<TValue, TError> {
    * @param {TValue} value The new value.
    * @param {FieldChangeHandlerContext<TError>} The context containing the validation result of the current value.
    */
-  onChange?: PickerChangeHandler;
+  onChange?: PickerChangeHandler<TValue, TError>;
```

Comment:
🙏

Reply:
I thought I did a PR when we found the issue weeks ago, I was surprised to have the same problem again :see_no_evil: